### PR TITLE
Improve support for cycles access

### PR DESCRIPTION
**Pull request by zhangsoledad**
### What problem does this PR solve?

Improve ckb's support for recording and accessing cycles as described in [3613](https://github.com/nervosnetwork/ckb/issues/3673), while implementing the fee_rate estimate short-term scheme.

### What is changed and how it works?

* For cycles and tx_size, the appending of records is done in a compatible way, so that when a new version of the node is run, new data is written, cycles and tx_size are appended, and old data is read in a compatible way. RPC returns data that is not recorded as `null`.
* `get_block`/`get_block_by_number` adds a parameter `with_cycles` to choose whether to return cycles or not, mainly because some scenarios, such as indexer, do not need cycles and are performance sensitive, so an additional parameter is added to avoid the extra overhead needed to return cycles.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
Title Only: Include only the PR title in the release note.
```

### CODE REVIEW

File: tx-pool/src/component/mod.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ const DEFAULT_BYTES_PER_CYCLES: f64 = 0.000_170_571_4_f64;
 
 /// Virtual bytes(aka vbytes) is a concept to unify the size and cycles of a transaction,
 /// tx_pool use vbytes to estimate transaction fee rate.
-pub(crate) fn get_transaction_virtual_bytes(tx_size: usize, cycles: u64) -> u64 {
```
Comment (yangby-cryptape) [inline tx-pool/src/component/mod.rs:23]: Could this function moved to `ckb-types`?

It's too heavy for third-party applications to call this function, they have to depend on `rocksdb`, `tokio` and so on.

**Review by driftluo — APPROVED** [review]

File: tx-pool/src/component/mod.rs (hunk 1 above)
Comment (zhangsoledad) [reply tx-pool/src/component/mod.rs:23]: It's fine to expose it. I didn't want to expose this function, it's so simple that there's no need to import it from `ckb-types`, other non-rust projects can't use it either.

Comment (yangby-cryptape) [reply tx-pool/src/component/mod.rs:23]: If expose it, at least, users could find several documentation lines.
It's really hard to find it.
Last week, a former ckb-core member had asked me for how to calculate `vbytes`.

If I was correct, there is no RFC mention there is a concept of "vbytes" in CKB and the magic number `DEFAULT_BYTES_PER_CYCLES`.
From normal users' point, they may not know there is a concept of "vbyes" in CKB.

And, no standard (RFC) means there is no promise whether the number `DEFAULT_BYTES_PER_CYCLES` would be changed, so depend on an official API is more safe than hard-code.

**Maybe we should standardize this concept, better than force users to read our code to find the magic number and the cryptic formula.**

### References

- [Section "Paying for Transaction Fees" in CKB RFC-0015](https://github.com/nervosnetwork/rfcs/blob/af5c8f32ef6142abe520c6e6cc688049db125f08/rfcs/0015-ckb-cryptoeconomics/0015-ckb-cryptoeconomics.md#paying-for-transaction-fees)
- [Section "Computation Cost and Transaction Fees" in CKB RFC-0002](https://github.com/nervosnetwork/rfcs/blob/af5c8f32ef6142abe520c6e6cc688049db125f08/rfcs/0002-ckb/0002-ckb.md#52-computation-cost-and-transaction-fees)

**Review by yangby-cryptape — APPROVED** [review]

**Pull request merged**